t/016-resp-header.t

```diff
@@ -1501,84 +1501,11 @@ test
 
 
 
-=== TEST 69: return the matched content-type instead of default_type
---- http_config
-types {
-    image/png png;
-}
---- config
-location /set/ {
-     default_type text/html;
-     content_by_lua_block {
-       ngx.say(ngx.header["content-type"])
-   }
-}
---- request
-GET /set/hello.png
---- response_headers
-Content-Type: image/png
---- response_body
-image/png
---- no_error_log
-[error]
-
-
-
-=== TEST 70: always return the matched content-type
---- config
-    location /set/ {
-        default_type "image/png";
-        content_by_lua_block {
-            ngx.say(ngx.header["content-type"])
-            ngx.say(ngx.header["content-type"])
-        }
-    }
---- request
-GET /set/hello.png
---- response_headers
-Content-Type: image/png
---- response_body
-image/png
-image/png
---- no_error_log
-[error]
```

Comment:
I think we should keep the above, but update them with the new conservative behavior we elaborated on. That is:

> Instead of overdoing, I think it would be better to return the real Content-Type header. When the header is missing, return nil instead of the default one.

Our other tests ensure that the response header has (or not) the Content-Type header, but those tests should ensure that in the _Lua-land_, the return values of those lookups is `nil`.

So we get:
```
=== TEST 70: return nil if Content-Type is not set
--- config
    location /set/ {
        default_type "image/png";
        content_by_lua_block {
            ngx.log(ngx.WARN, "Content-Type: ", h["content-type"])
            ngx.say("Content-Type: ", ngx.header["content-type"])
        }
    }
--- request
GET /set/hello.png
--- response_headers
!Content-Type
--- response_body
Content-Type: nil
--- no_error_log
[error]
--- error_log
Content-Type: nil
```

Also, this test should be moved below, where other CT-related tests are.

Reply:
Test added in an updated version of the patch.